app/utils/transformers/AdditionalInformationTransformer.scala

```diff
@@ -59,4 +55,37 @@ class AdditionalInformationTransformer @Inject() (referenceDataConnector: Refere
       })
     }
   }
+
+  def transform(additionalReferences: Seq[AdditionalInformationType02], hcIndex: Index, itemIndex: Index)(implicit
+    hc: HeaderCarrier
+  ): UserAnswers => Future[UserAnswers] = userAnswers => {
+
+    lazy val referenceDataLookups = additionalReferences.map {
+      additionalInformation =>
+        referenceDataConnector
+          .getAdditionalInformationCode(additionalInformation.code)
+          .map(TempAdditionalInformation(additionalInformation, _))
+    }
+
+    Future.sequence(referenceDataLookups).flatMap {
+      _.zipWithIndex.foldLeft(Future.successful(userAnswers))({
+        case (acc, (TempAdditionalInformation(underlying, additionalInformation), i)) =>
+          acc.flatMap {
+            userAnswers =>
+              val index = Index(i)
+              val pipeline: UserAnswers => Future[UserAnswers] = {
+                set(HouseConsignmentAdditionalInformationCodePage(hcIndex, itemIndex, index), additionalInformation) andThen
```

Comment:
Can we also set the sequence number here? See other transformers for examples :)

Reply:
Refactored to have sequence number now